lib/active_merchant/billing/gateways/visanet_peru.rb

```diff
@@ -34,25 +34,33 @@ def authorize(amount, payment_method, options={})
         params[:email] = options[:email] || 'unknown@email.com'
         params[:createAlias] = false
 
-        commit("authorize", params)
+        commit("authorize", params, options)
       end
 
       def capture(authorization, options={})
         params = {}
+        options[:id_unico] = parse_auth(authorization)[1]
         add_auth_order_id(params, authorization, options)
-        commit("deposit", params)
+        commit("deposit", params, options)
       end
 
       def void(authorization, options={})
         params = {}
         add_auth_order_id(params, authorization, options)
-        commit("void", params)
+        commit("void", params, options)
       end
 
       def refund(amount, authorization, options={})
+        # Attempt RefundSingleTransaction if unsettled
         params = {}
+        params[:amount] = amount(amount)
```

Comment:
Should this only be `if amount`, since it may not be provided if it's a full refund, and sending it as nil might cause an issue?

Reply:
I haven't touched this. Seems like its currently not causing an issue.

Reply:
Hmm it looks like this was modified to pass in the amount. The current `refund` method on master ignores the amount: https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/visanet_peru.rb#L52-L56. I'm thinking we'll need a guard around the `amount` here.

Reply:
@curiousepic sorry I thinking about the `params[:amount]` in add_invoice. Updated to have the if statement.